magtogoek/adcp/process.py

```diff
@@ -670,12 +670,24 @@ def _regrid_dataset(dataset: xr.Dataset, pconfig: ProcessConfig) -> xr.Dataset:
     and with equal weights for all data inside each bin.
  
     """
+    # Apply quality control
+    for var_ in 'uvw':
+        dataset[var_] = dataset[var_].where(dataset[f"{var_}_QC"] < 2)
+
+    # Log entry
     msg = 'to grid from file: %s' % pconfig.grid_depth
     l.log(f"Regridded dataset with method {pconfig.grid_method} {msg}")
-    return regrid_dataset(dataset,
-                          grid=pconfig.grid_depth,
-                          dim='depth',
-                          method=pconfig.grid_method)
+
+    # Regridding
+    dataset = regrid_dataset(dataset,
```

Comment:
I think we now have a problem witht the flag values. How do you average flag values ?